pyard/data_repository.py

```diff
@@ -325,25 +325,23 @@ def generate_alleles_and_xx_codes_and_who(db_connection: sqlite3.Connection, img
     shortnulls = dict()
     for who in who_group:
         # e.g. DRB4*01:03
-        expression_alleles=[]
-        expression_chars_found = set()
+        expression_alleles = dict()
```

Comment:
You should use a `defaultdict` with `list` in cases where you need an empty list when there's no key.

```python
from collections import defaultdict
...
expression_alleles = defaultdict(list)
```